android/java/org/chromium/chrome/browser/settings/BraveVpnPreferences.java

```diff
@@ -477,20 +489,23 @@ private void showConfirmDialog() {
                 getActivity().getResources().getString(R.string.reset_vpn_config_dialog_message));
         confirmDialog.setPositiveButton(
                 getActivity().getResources().getString(android.R.string.yes), (dialog, which) -> {
-                    BraveVpnNativeWorker.getInstance().invalidateCredentials(
-                            BraveVpnPrefUtils.getHostname(), BraveVpnPrefUtils.getClientId(),
-                            BraveVpnPrefUtils.getSubscriberCredential(),
-                            BraveVpnPrefUtils.getApiAuthToken());
-                    BraveVpnUtils.showProgressDialog(
-                            getActivity(), getResources().getString(R.string.resetting_config));
-                    new Handler().postDelayed(() -> {
-                        BraveVpnUtils.resetProfileConfiguration(getActivity());
-                        new Handler().post(() -> updateSummaries());
-                    }, INVALIDATE_CREDENTIAL_TIMER_COUNT);
+                    resetConfiguration();
                     dialog.dismiss();
                 });
         confirmDialog.setNegativeButton(getActivity().getResources().getString(android.R.string.no),
                 (dialog, which) -> { dialog.dismiss(); });
         confirmDialog.show();
     }
+
+    private void resetConfiguration() {
+        BraveVpnNativeWorker.getInstance().invalidateCredentials(BraveVpnPrefUtils.getHostname(),
+                BraveVpnPrefUtils.getClientId(), BraveVpnPrefUtils.getSubscriberCredential(),
+                BraveVpnPrefUtils.getApiAuthToken());
+        BraveVpnUtils.showProgressDialog(
+                getActivity(), getResources().getString(R.string.resetting_config));
+        new Handler().postDelayed(() -> {
+            BraveVpnUtils.resetProfileConfiguration(getActivity());
+            new Handler().post(() -> updateSummaries());
+        }, INVALIDATE_CREDENTIAL_TIMER_COUNT);
+    }
```

Comment:
this can produce leak or crash, must check if the fragment is active, can also clear this in onpause etc.